### DISCUSSION TITLE
Error: deck.gl: assertion failed (Console Flooded)

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#5638] [Q&A] [answered]
I am using deck v 8.4.11 and have started noticing my console being flooded with deck errors stating...

```
error during initialization of ScatterplotLayer({id: 'layer id'}) Error: deck.gl: assertion failed.
at assert (assert.js?54c0:3)
at ScatterplotLayer._initState (layer.js?2bb8:1068)
at ScatterplotLayer._initialize (layer.js?2bb8:731)
```

In layer.js there is an assertion test in the `_initState` function that appears to be failing...

`assert(!this.internalState && !this.state)`

Initially I thought it was because the my layers are `CompositeLayers `and they were not implementing the `initializeState` function. So, it appeared as though the assertion was failing on the !this.state.

However, I implemented the initializeState as a test like so...

```
initializeState() {
    this.state = { test: 'test' };
}

```
This however did not correct the issue. The assertion condition `!this.internalState && !this.state` still fails. Both `this.internalState` and `this.state` objects have values so it is unclear to me why the assertion would be failing.

Additionally, the layers this assertion is failing on are updating at a high rate so this ends up flooding my console and ultimately appears to cause Chrome to crash.

**Comment by Pessimistress** [comment, marked as answer]
You get this error because you are reusing a layer instance that has been finalized. See discussion in #4176.

**Reply by jfuehner** [reply]
Thank you! This was super helpful. I was able to track down where in our code layers were being reused.

**Comment by AmarnathV7867** [comment]
I'm facing similar issue can anyone of you connect with me through google meet, please give me solution for similar issue
